src/command.rs

```diff
@@ -175,72 +198,118 @@ where
         }
     }
 
-    fn insert(self, world: &mut World)
+    fn build(self, buffer: &mut CommandBuffer)
     where
         <TS as ConsFlatten>::Output: TagSet + TagLayout + for<'a> Filter<ChunksetFilterData<'a>>,
         ComponentTupleSet<
             <CS as ConsFlatten>::Output,
             std::iter::Once<<CS as ConsFlatten>::Output>,
         >: ComponentSource,
     {
-        world.insert_buffered(
-            self.entity,
-            self.tags.flatten(),
-            std::iter::once(self.components.flatten()),
-        );
+        buffer
+            .commands
+            .get_mut()
+            .push(EntityCommand::WriteWorld(Arc::new(InsertCommand {
+                write_components: Vec::default(),
+                write_tags: Vec::default(),
+                tags: self.tags.flatten(),
+                components: IntoComponentSource::into(std::iter::once(self.components.flatten())),
+                entities: vec![self.entity],
+            })));
     }
 }
 
+#[derive(Debug)]
+pub enum CommandError {
+    EntityBlockFull,
+}
+impl std::fmt::Display for CommandError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result { write!(f, "CommandError") }
+}
+
+impl std::error::Error for CommandError {
+    fn description(&self) -> &str { "CommandError" }
```

Comment:
This method is soft-deprecated, and should not be implemented? See the bold text on https://doc.rust-lang.org/std/error/trait.Error.html#method.description

Reply:
Fixed. Thanks for that.